plugins/inputs/nftables/nftables_test.go

```diff
@@ -0,0 +1,225 @@
+                  "protocol": "tcp",
+                  "field": "dport"
+                }
+              },
+              "right": 22
+            }
+          },
+          {
+            "accept": null
+          }
+        ]
+      }
+    },
+    {
+      "rule": {
+        "family": "inet",
+        "table": "test",
+        "chain": "test-chain",
+        "handle": 2,
+        "comment": "test1",
+        "expr": [
+          {
+            "match": {
+              "op": "==",
+              "left": {
+                "payload": {
+                  "protocol": "tcp",
+                  "field": "dport"
+                }
+              },
+              "right": 22
+            }
+          },
+          {
+            "counter": {
+              "packets": 1,
+              "bytes": 2 
+            }
+          },
+          {
+            "accept": null
+          }
+        ]
+      }
+    },
+    {
+      "rule": {
+        "family": "inet",
+        "table": "test",
+        "chain": "no_comment",
+        "handle": 2,
+        "expr": [
+          {
+            "match": {
+              "op": "==",
+              "left": {
+                "payload": {
+                  "protocol": "tcp",
+                  "field": "dport"
+                }
+              },
+              "right": 22
+            }
+          },
+          {
+            "accept": null
+          }
+        ]
+      }
+    },
+    {
+      "rule": {
+        "family": "inet",
+        "table": "test",
+        "chain": "test-chain",
+        "handle": 2,
+        "comment": "test2",
+        "expr": [
+          {
+            "match": {
+              "op": "==",
+              "left": {
+                "payload": {
+                  "protocol": "tcp",
+                  "field": "dport"
+                }
+              },
+              "right": 22
+            }
+          },
+          {
+            "counter": {
+              "packets": 24468,
+              "bytes": 1412296
+            }
+          },
+          {
+            "accept": null
+          }
+        ]
+      }
+    }
+  ]
+}`
+
+func TestParseNftableBadRule(t *testing.T) {
+	badrules := []string{
+		`{ "nftables": [
+    {
+      "rule": "bad"
+    }
+  ]
+}`,
+		`{ "nftables": [
+    {
+      "rule": {}
+    },
+}`,
+		`{ "nftables": [
+    {
+      "rule": []
+    },
+}`}
+	baderrs := []string{
+		"error parsing: { \"nftables\": [\n    {\n      \"rule\": \"bad\"\n    }\n  ]\n}, Error: unable to parse rule: unable to unmarshal: \"bad\"",
+		"error parsing: { \"nftables\": [\n    {\n      \"rule\": {}\n    },\n}, Error: invalid character '}' looking for beginning of value",
+		"error parsing: { \"nftables\": [\n    {\n      \"rule\": []\n    },\n}, Error: invalid character '}' looking for beginning of value",
+	}
+	for i, v := range badrules {
+		acc := new(testutil.Accumulator)
+		err := parseNftableOutput([]byte(v), acc)
+		if err.Error() != baderrs[i] {
```

Comment:
Instead of using custom error matching, use require package patterns, see example in other tests.